src/persist-client/src/write.rs

```diff
@@ -219,10 +219,25 @@ where
                     if PartialOrder::less_than(&current_upper, &lower) {
                         self.upper = current_upper.clone();
                         return Ok(Ok(Err(Upper(current_upper))));
+                    } else if PartialOrder::less_equal(&current_upper, desc.upper()) {
```

Comment:
isn't the equal case also a no-op?

Reply:
yeah, understood! my question is whether this line should be `less_than` instead of `less_equal`. in that framing, I think `current_upper == desc.upper()` is case 3, no?

Reply:
> ((I think the current code wasn't incorrect, it was just doing potentially unnecessary work.))

yeah that's my model of it as well